base/src/external_transition.rs

```diff
@@ -10,11 +10,11 @@ use crate::types::*;
 #[serde(from = "<Self as WireType>::WireType")]
 #[serde(into = "<Self as WireType>::WireType")]
 pub struct ExternalTransition {
-    protocol_state: ProtocolState,
-    protocol_state_proof: ProtocolStateProof,
-    staged_ledger_diff: StagedLedgerDiff,
-    delta_transition_chain_proof: DeltaTransitionChainProof,
-    current_protocol_version: ProtocolVersion,
-    proposed_protocol_version_opt: Option<ProtocolVersion>,
-    validation_callback: (),
+    pub protocol_state: ProtocolState,
```

Comment:
Any reason we need to make all these public? This isn't clear to me from reading the PR

Reply:
Plan to add more testcase on specific fields

Reply:
We really want this hey https://internals.rust-lang.org/t/pub-test-visibility-modifier/12587
Shame it doesn't exist yet